crypto/src/merkle/mod.rs

```diff
@@ -86,8 +86,8 @@ mod tests;
 /// ```
 #[derive(Debug)]
 pub struct MerkleTree<H: Hasher> {
-    nodes: Vec<H::Digest>,
-    leaves: Vec<H::Digest>,
+    pub nodes: Vec<H::Digest>,
+    pub leaves: Vec<H::Digest>,
 }
```

Comment:
I don't mind this change, but I wonder if instead we could meet the requirements by adding another constructor. Maybe something like:
```Rust
pub fn from_raw_parts(nodes: Vec<H::Digest>, leaves: Vec<H::Digest>) -> Self
```